sdk/src/clearingHouse.ts

```diff
@@ -611,32 +801,88 @@ export class ClearingHouse {
 	 */
 	public async initializeUserAccountAndDepositCollateral(
 		amount: BN,
-		userTokenAccount: PublicKey,
-		bankIndex = new BN(0),
+		authority: PublicKey,
```

Comment:
I think we should still pass in the token address here. A token account isn't guaranteed to be an associated and we should let the user pass in a non associated token account. 

If the user doesn't have an WSOL token account, we can just let them pass in their authority instead of the token account and the function can handle make the temp token account for them

I also don't want to have to change all the tests to pass in authority instead of take account